common/src/main/java/org/apache/falcon/entity/ClusterHelper.java

```diff
@@ -71,6 +73,18 @@ public static Configuration getConfiguration(Cluster cluster) {
         return conf;
     }
 
+    public static Configuration getConfiguration(String storageUrl, String executeEndPoint,
+                                                 String kerberosPrincipal) {
+        Configuration conf = new Configuration();
+        conf.set(HadoopClientFactory.FS_DEFAULT_NAME_KEY, storageUrl);
+        conf.set(HadoopClientFactory.MR_JT_ADDRESS_KEY, executeEndPoint);
+        conf.set(HadoopClientFactory.YARN_RM_ADDRESS_KEY, executeEndPoint);
+        if (StringUtils.isNotBlank(kerberosPrincipal)) {
```

Comment:
We cannot assume that the falcon server and the cluster have same security level. Falcon server can be unsecure, and cluster can be secure.